### PR TITLE
Exclude file path

### PR DESCRIPTION
**Pull request by vrischmann**
This PR adds a search field to exclude file paths with a regexp. It works exactly like the current file path search, except if it matches it rejects the file.

This is the same as #223 but because I had removed my hound fork a long time ago I'm unable to update that PR.

I rebased on the latest master and did a quick test that it still works.

### CODE REVIEW

File: index/index.go [hunk 1]
```diff
@@ -167,6 +168,14 @@ func (n *Index) Search(pat string, opt *SearchOptions) (*SearchResponse, error)
 		}
 	}
 
+	var efre *regexp.Regexp
```
Comment (jacobrose) [inline index/index.go:171]: I think our variable names are getting a tad too cryptic -- what do you think about `excludeFre`?

**Review by jacobrose — COMMENTED** [review]
Hi, @vrischmann! Great contribution, one which we've had requests for. I left a comment for you to consider including in your PR, but we're ready to approve either way. Please let me know if you'd like to make that change before merging. Thanks again!

File: index/index.go (hunk 1 above)
Comment (vrischmann) [reply index/index.go:171]: Sure, I'll rename it.

**Review by jacobrose — APPROVED** [review]
Thank you so much, @vrischmann!

File: ui/assets/js/hound.js [hunk 2]
```diff
@@ -373,6 +374,23 @@ var SearchBar = React.createClass({
   filesGotFocus: function(event) {
     this.showAdvanced();
   },
+  excludeFilesGotKeydown: function(event) {
+    switch (event.keyCode) {
+    case 38:
+      // if advanced is empty, close it up.
+      if (this.refs.excludeFiles.getDOMNode().value.trim() === '') {
```
Comment (salemhilal) [inline ui/assets/js/hound.js:381]: I think this check made sense when there was only one text field, but now that there's two, you'll likely have to check that both this text field and the `this.refs.files` text field are empty or you'll end up closing the advanced view when it's still in use.

**Review by salemhilal — COMMENTED** [review]
Sorry for the slow back-and-forth. I have one comment, but this looks great (and super straightforward).

File: ui/assets/js/hound.js (hunk 2 above)
Comment (vrischmann) [reply ui/assets/js/hound.js]: Okay, I addressed that in the latest commit.

**Review by salemhilal — APPROVED** [review]
Wicked fast, thank you!

**Comment by salemhilal** [comment]
Gave this branch one last local test, and everything works as expected.
Thanks again for reviving such an old branch @vrischmann. Your work is super appreciated.

**Pull request merged**